src/xscen/regrid.py

```diff
@@ -350,54 +372,104 @@ def _regridder(
         unmapped_to_nan=unmapped_to_nan,
         **kwargs,
     )
-    if not Path(filename).is_file():
+    if filename is not None and not Path(filename).is_file():
         regridder.to_netcdf(filename)
 
     return regridder
 
 
-def create_bounds_rotated_pole(ds: xr.Dataset):
+def create_bounds_rotated_pole(ds: xr.Dataset) -> xr.Dataset:
+    warnings.warn(
+        "This function is deprecated and will be removed in xscen v0.12.0. Use create_bounds_gridmapping instead.",
+        FutureWarning,
+    )
+    return create_bounds_gridmapping(ds, "rotated_pole")
+
+
+def create_bounds_gridmapping(ds: xr.Dataset, gridmap: str) -> xr.Dataset:
     """Create bounds for rotated pole datasets."""
-    ds = ds.cf.add_bounds(["rlat", "rlon"])
+    xname = ds.cf.axes["X"][0]
+    yname = ds.cf.axes["Y"][0]
+
+    ds = ds.cf.add_bounds([yname, xname])
 
     # In "vertices" format then expand to 2D. From (N, 2) to (N+1,) to (N+1, M+1)
-    rlatv1D = cfxr.bounds_to_vertices(ds.rlat_bounds, "bounds")
-    rlonv1D = cfxr.bounds_to_vertices(ds.rlon_bounds, "bounds")
-    rlatv = rlatv1D.expand_dims(rlon_vertices=rlonv1D).transpose(
-        "rlon_vertices", "rlat_vertices"
-    )
-    rlonv = rlonv1D.expand_dims(rlat_vertices=rlatv1D).transpose(
-        "rlon_vertices", "rlat_vertices"
+    yv1D = cfxr.bounds_to_vertices(ds[f"{yname}_bounds"], "bounds")
+    xv1D = cfxr.bounds_to_vertices(ds[f"{xname}_bounds"], "bounds")
+    yv = yv1D.expand_dims(dict([(f"{xname}_vertices", xv1D)])).transpose(
+        f"{xname}_vertices", f"{yname}_vertices"
     )
-    central = ds.rotated_pole.attrs.get("north_pole_grid_longitude")
-    central = float(central) if central is not None else None
-    # Get cartopy's crs for the projection
-    RP = ccrs.RotatedPole(
-        pole_longitude=float(ds.rotated_pole.grid_north_pole_longitude),
-        pole_latitude=float(ds.rotated_pole.grid_north_pole_latitude),
-        central_rotated_longitude=central,
+    xv = xv1D.expand_dims(dict([(f"{yname}_vertices", yv1D)])).transpose(
+        f"{xname}_vertices", f"{yname}_vertices"
     )
+
+    if gridmap == "rotated_pole":
+        central = ds.rotated_pole.attrs.get("north_pole_grid_longitude")
+        central = float(central) if central is not None else None
+        # Get cartopy's crs for the projection
+        RP = ccrs.RotatedPole(
+            pole_longitude=float(ds.rotated_pole.grid_north_pole_longitude),
+            pole_latitude=float(ds.rotated_pole.grid_north_pole_latitude),
+            central_rotated_longitude=central,
+        )
+    elif gridmap == "oblique_mercator":
+        RP = ccrs.ObliqueMercator(
+            central_longitude=float(ds.oblique_mercator.longitude_of_projection_origin),
+            central_latitude=float(ds.oblique_mercator.latitude_of_projection_origin),
+            false_easting=float(ds.oblique_mercator.false_easting),
+            false_northing=float(ds.oblique_mercator.false_northing),
```

Comment:
Ces deux-là sont marqués comme optionnel dans la doc de CF, ce serait bien de faire comme en haut? Ou en fait:
```suggestion
            false_easting=float(ds.oblique_mercator.attrs.get('false_easting', 0)),
            false_northing=float(ds.oblique_mercator.attrs.get('false_northing', 0)),
```

Reply:
Je ne voulais pas mettre de valeur juste au cas où le défaut change plus tard. J'ai créé une petite sous-fonction pour que ça soit plus clean.